### PR TITLE
Prune deprecated Trainer(checkpoint_callback=ModelCheckpoint())

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Removes `Trainer(checkpoint_callback=ModelCheckpoint())`

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Removed
 
 
+- Removed passing a `ModelCheckpoint` instance to `Trainer(checkpoint_callback)` ([#6165](https://github.com/PyTorchLightning/pytorch-lightning/pull/6165))
```
Comment (carmocca) [inline CHANGELOG.md:22]:
```suggestion
- Removed passing a `ModelCheckpoint` instance to `Trainer(checkpoint_callback)` ([#6166](https://github.com/PyTorchLightning/pytorch-lightning/pull/6166))
```

**Review by s-rog — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Comment by edenlightning** [comment]
do we need to also change https://github.com/PyTorchLightning/pytorch-lightning/blob/fc9bb53e13edaeb48e61ac535d129279d04cb2ca/tests/accelerators/test_tpu_backend.py#L49?

**Comment by carmocca** [comment]
No, this only removes support for passing a `ModelCheckpoint` instance. Passing a `bool` is still allowed

**Review by awaelchli — APPROVED** [review]
Thanks!

**Review by tchaton — APPROVED** [review]
